src/PreCICE.jl

```diff
@@ -1332,5 +1340,149 @@ function actionReadIterationCheckpoint()
     return unsafe_string(msgCstring)
 end
 
+@doc """
+
+        isGradientDataRequired(dataID::Integer)
+        
+Checks if the given data set requires gradient data. We check if the data object has been intialized with the gradient flag.
+# Arguments
+- `dataID::Integer`: ID of the data to be checked. Obtained by [`getDataID`](@ref).
+
+"""
+function isGradientDataRequired(dataID::Integer)
+    return ccall((:precicec_isGradientDataRequired, "libprecice"), Cbool, dataID)
```

Comment:
We use int instead of bool for the julia bindings as C does not have a bool type.
Try:
```
    return ccall((:precicec_isGradientDataRequired, "libprecice"), Cint, (Cint,), dataID)
```
Which fixed all errors for me